it/controllers/agent/incomeSources/add/AddBusinessTradeControllerISpec.scala

```diff
@@ -0,0 +1,187 @@
+
+  val addBusinessAddressUrl = controllers.incomeSources.add.routes.AddBusinessAddressController.showAgent(isChange = false).url
+  val incomeSourcesUrl: String = controllers.routes.HomeController.showAgent.url
+  val checkDetailsUrl: String = controllers.incomeSources.add.routes.CheckBusinessDetailsController.showAgent().url
+
+
+  val pageTitleMsgKey: String = messagesAPI("add-business-trade.heading")
+  val pageHint: String = messagesAPI("add-business-trade.p1")
+  val button: String = messagesAPI("base.continue")
+
+
+  s"calling GET $addBusinessTradeControllerShowUrl" should {
+    "render the Add Business trade page for an Agent" when {
+      "User is authorised" in {
+        Given("I wiremock stub a successful Income Source Details response with multiple businesses and a uk property")
+        stubAuthorisedAgentUser(true)
+        enable(IncomeSources)
+        IncomeTaxViewChangeStub.stubGetIncomeSourceDetailsResponse(testMtditid)(OK, multipleBusinessesResponse)
+        When(s"I call GET $addBusinessTradeControllerShowUrl")
+        val res = IncomeTaxViewChangeFrontend.getAddBusinessTrade(clientDetailsWithConfirmation)
+        verifyIncomeSourceDetailsCall(testMtditid)
+
+        res should have(
+          httpStatus(OK),
+          pageTitleAgent(pageTitleMsgKey),
+          elementTextByID("business-trade-hint")(pageHint),
+          elementTextByID("continue-button")(button)
+        )
+      }
+    }
+    "303 SEE_OTHER - redirect to home page" when {
+      "Income Sources FS disabled" in {
+        stubAuthorisedAgentUser(true)
+
+        Given("I wiremock stub a successful Income Source Details response with no businesses or properties")
+        disable(IncomeSources)
+        IncomeTaxViewChangeStub.stubGetIncomeSourceDetailsResponse(testMtditid)(OK, noPropertyOrBusinessResponse)
+
+        When(s"I call GET ${addBusinessTradeControllerShowUrl}")
+        val result = IncomeTaxViewChangeFrontend.getAddBusinessTrade(clientDetailsWithConfirmation)
+
+        result should have(
+          httpStatus(SEE_OTHER),
+          redirectURI(incomeSourcesUrl)
+        )
+      }
+    }
+  }
+
+  s"calling POST ${addBusinessTradeSubmitUrl}" should {
+    s"303 SEE_OTHER and redirect to $addBusinessAddressUrl" when {
+      "User is authorised and business trade is valid" in {
+        stubAuthorisedAgentUser(true)
+        enable(IncomeSources)
+        IncomeTaxViewChangeStub.stubGetIncomeSourceDetailsResponse(testMtditid)(OK, noPropertyOrBusinessResponse)
+
+        val formData: Map[String, Seq[String]] = {
+          Map(
+            BusinessTradeForm.businessTrade -> Seq("Test Business Trade")
+          )
+        }
+
+        When(s"I call POST ${addBusinessTradeSubmitUrl}")
+        val result = IncomeTaxViewChangeFrontend.post("/income-sources/add/business-trade", clientDetailsWithConfirmation)(formData)
+        result should have(
```

Comment:
Can you add a check for business trade name ? check with database using the session service